patterns/builder.md

```diff
@@ -2,46 +2,58 @@
 
 ## Description
 
-Construct an object with calls to a builder helper. 
-
+Construct an object with calls to a builder helper.
 
 ## Example
 
 ```rust
-struct Foo {
+#[derive(Debug)]
+pub struct Foo {
     // Lots of complicated fields.
+    bar: String,
+}
+
+impl PartialEq for Foo {
+    fn eq(&self, other: &Self) -> bool {
+        self.bar == other.bar
+    }
 }
 
-struct FooBuilder {
+pub struct FooBuilder {
     // Probably lots of optional fields.
-    ...
+    bar: String,
 }
 
 impl FooBuilder {
-    fn new(...) -> FooBuilder {
+    pub fn new() -> FooBuilder {
         // Set the minimally required fields of Foo.
+        FooBuilder {
+            bar: "X".to_string(),
+        }
     }
 
-    fn named(mut self, name: &str) -> FooBuilder {
+    pub fn name(mut self, bar: &str) -> FooBuilder {
         // Set the name on the builder itself, and return the builder by value.
+        self.bar = bar.to_string();
```

Comment:
Is this a good pattern to clone within the function when users can provide `String`? I am not quite sure how this is done in the wild.